### PR TITLE
fix: GH-9103 Use DatePickerFormatterService in Replenishment Order

### LINKED ISSUE #9103: Use DatePickerFormatterService in Replenishment Order, and also fix the logic if it applies.
master ticket: GH-8202

* This will be worked on after merging epic/b2b-scheduled-repl into develop

Similarly, replenishment was using `toISOString` to convert the date into date format of ISO 8601.
Therefore, replenishment should start using DatePickerFormatterService services

---
Additionally, the logic for both is not the correct one as it is converting the local date time (your current timezone) into the general UTC+0 date. I noticed sometimes when ordering a replenishment order that the date did not match as the conversion from local date to UTC broke it. --> To verify with any others

### PR DESCRIPTION
**Pull request by saad-mo**
closes #9103 

- Moved `DatePickerFormatterService` to `core/src/util` since `storefrontlib` has a dependency o `core` and not the other way around
- Added a new method `toISOString` to accommodate the format required in `OccReplenishmentOrderFormSerializer` 
- Udpated `OccReplenishmentOrderFormSerializer` to use `DatePickerFormatterService` 
- Updated unit tests accordingly

### CODE REVIEW

File: projects/core/src/util/date-picker-formatter.service.ts [hunk 1]
```diff
@@ -23,4 +23,9 @@ export class DatePickerFormatterService {
       return date;
     }
   }
+
+  toISOString(value: string): string {
+    const dateTime = '00:00:00';
+    return value ? this.toNative(value) + 'T' + dateTime + 'Z' : null;
```
Comment (bgambocjaviniar) [inline projects/core/src/util/date-picker-formatter.service.ts]: I have mentioned that the date logic is wrong in replenishment in the issue ticket that was raised, which is why we need to use re-use the service that came from my-company, however, we need to check if the logic is fine as well and fix or create things accordingly.

For example of why this logic doesn't work, try placing a replenishment for January 1st 2021, and you will see that the returned date is December 31st 2020, which is wrong.

Moreover, `toModel` function is doing the same thing as this imported function from serializer, when a user chooses a date from the date picker.

**Review by bgambocjaviniar — CHANGES_REQUESTED** [review]
- Moved `DatePickerFormatterService` to `core/src/util` since `storefrontlib` has a dependency o `core` and not the other way around --> 🟢 
- Added a new method `toISOString` to accommodate the format required in `OccReplenishmentOrderFormSerializer` --> 🔴   **as mentioned in the issue, this is the problem at hand as the date logic is wrong. For example, placing an order for January 1, 2011 will display `next order date` as December 31, 2020. Read potential solution below.**
- Udpated `OccReplenishmentOrderFormSerializer` to use `DatePickerFormatterService` --> 🟡 **if we're bringing the date*service, then shouldn't we bring the other service as well? `DateTimePickerFormatterService`. What do you think?** 
- Updated unit tests accordingly  --> 🔴 **, as mentioned, functionality is wrong.**

Problems --> please re-read the issue
1 - date logic is wrong because it's placing an order in UTC+0 even if you're in a different timezone. We need to accommodate the client's date, time, and timezone (offset) 

POTENTIAL solution
- re-verify the services functionality and fix things accordingly to accommodate the date. But make sure logic is fine for all.
- create a new function that creates the format that we want of ISO 8601 with the date + datetime and offset of the user's client date. Moreover, remove the `existing` functions **if needed** as we don't want UTC+0 date. **Therefore, please verify with a pair for a solution or ask the person who created these functionality on what was the use case for the services, and plan accordinly. (do note that their format is also 8601)**

File: projects/core/src/util/date-picker-formatter.service.ts (hunk 1 above)
Comment (Zeyber) [reply projects/core/src/util/date-picker-formatter.service.ts]: To shed some light on what is happening with the DatePicker, it uses the native `datetime-local` input element which will automatically append the timezone information (eg. CEST +2) when the datetime value is submitted in a request to the backend.

The backend cannot store this timezone information and will store the time in UTC-0 and adding/subtracting timezone offset hours to the final value stored.

If we display this date again from a backend response and put the value through a date pipe such at `cxDate`, the local timezone is used to convert the date value from the backend to the time relevant to the client (ie. local time offset value is displayed).

PS. I recall an issue with daylight savings time where the wrong offset is used if you are currently in/using a local timezone in daylight savings and the date you are setting to is not. Incorrect dates can be displayed or submitted to the backend by an hour. This was ignored at the time as an hour's discrepancy does not make the component unusable but I suspect that is what's happening with this issue.

Comment (saad-mo) [reply projects/core/src/util/date-picker-formatter.service.ts]: @bgambocjaviniar @Zeyber  

Thanks for the input. I will create a fresh PR and add detailed comments and findings.

**Comment by saad-mo** [comment]
Fresh PR #9457

**Pull request closed without merging**